### DISCUSSION TITLE
No Flannel, Kube router, Helm controller on fresh install

### DISCUSSION THREAD

**Discussion by NiklasRosenstein** [k3s-io/k3s#7027] [Q&A] [answered]
**Environmental Info:**
K3s Version: v1.25.6+k3s1 (9176e03c)

Node(s) CPU architecture, OS, and Version: x86_64, Ubuntu 22.04

Cluster Configuration: 1 Server

**Describe the bug:** I feel like I'm missing something, but every place I look it says K3s comes with Flannel, Kube router and Helm controller preinstalled. But on a fresh 1 server installation of K3s, I cannot find these three.

```
$ curl -sfL https://get.k3s.io | sh -
$ kubectl get pod --all-namespaces
NAMESPACE     NAME                                      READY   STATUS      RESTARTS   AGE
kube-system   coredns-597584b69b-92rwn                  1/1     Running     0          36m
kube-system   local-path-provisioner-79f67d76f8-pklww   1/1     Running     0          36m
kube-system   helm-install-traefik-crd-nclq6            0/1     Completed   0          36m
kube-system   svclb-traefik-a25faa28-whqpg              2/2     Running     0          35m
kube-system   helm-install-traefik-2ndwv                0/1     Completed   1          36m
kube-system   traefik-66c46d954f-jzppl                  1/1     Running     0          35m
kube-system   metrics-server-5f9f776df5-lxcvc           1/1     Running     0          36m
$ kubectl get daemonset --all-namespaces
NAMESPACE     NAME                     DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
kube-system   svclb-traefik-a25faa28   1         1         1       1            1           <none>          37m
$ ps aux | grep flanneld
root        4744  0.0  0.1   6608  2364 pts/0    S+   01:51   0:00 grep --color=auto flanneld
```

I've tested this also on a 1-server-2-worker configuration and got the same result. Pod-to-Pod communication was only working if both Pods were running on the same Node until I manually installed Flannel. There doesn't seem to be an "enable" flag that I've missed. I've tried explicitly setting `--flannel-backend vxlan` to no avail.

**Steps To Reproduce:** Install K3s and check the installed daemonsets/deployments.

**Comment by brandond** [comment, marked as answer]
K3s runs most of the core components in the main process, so you won't find pods for any of the things you're looking for. 

If networking between pods is not working, make sure that the correct ports are open between your nodes. If your nodes are not on the same local network, you may need to use a different flannel backend than vxlan.

**Reply by NiklasRosenstein** [reply]
Thanks for the response @brandond. Is there a way for me to check the Flannel configuration and logs that is running inside the K3s process?

Is there a place where it is documented which of the components are actually embedded in the K3s process? Or something like a `k3s status` and it would tell me `flannel: ok`, and vice versa?

I've found references to k8s, kubectl, etcd, containerd and klipper-lb as being embedded in various places (e.g. README or the docs), but nothing about these other three components.

**Reply by NiklasRosenstein** [reply]
I believe I found the issue: I needed to set `--flannel-iface ens10` because that is the interface for the private network but it was not the first. However, I had it set only on the server and not on the agents. Also setting it on the agents fixed it.

**Reply by brandond** [reply]
Yes, interfaces configured are per host, since it's not expected that all nodes would have the exact same interfaces.

> Is there a place where it is documented which of the components are actually embedded in the K3s process? Or something like a k3s status and it would tell me flannel: ok, and vice versa?

Not really. Anything that doesn't have a pod is running in the main process. There no command to check status; if any of the components fail the entire process aborts so that everything can be reinitialized.
